libmachine/drivers/base.go

```diff
@@ -18,8 +18,8 @@ type VMDriver struct {
 	*BaseDriver
 	ImageSourcePath string
 	ImageFormat     string
-	Memory          int
-	CPU             int
+	Memory          uint64
```

Comment:
I'd use `uint` for memory as it was an `int` before.

Reply:
@cfergeau but in the crc side most of the place it is converted to `uint64` because it is changed to bytes from GB.

Reply:
The GB to bytes conversion in https://github.com/praveenkumar/crc/commit/ef1b33c3fd99c0f8d872c68d8dd968d17b068193 does not appear to have moved to a different place, so uint to hold GB value, and uint64 to hold bytes value should work?
 `Memory` and `DiskSize` would even be better as `github.com/containers/common/pkg/strongunits` but that's likely to be quite some work.

Reply:
>  uint to hold GB value, and uint64 to hold bytes value should work?

Or does this require to keep a lot of casts?

Reply:
@cfergeau it will require still require to cast `uint` to `uint64` for `ConvertGiBToBytes` function which return to `uint64` and that is because `driver.DiskCapacity` accept `uint64` . If we even do single conversion then also golint going to complain and we might need to ignore it by adding the comment but this whole exercise is to avoid ignoring golint complain.

Reply:
`ConvertGiBToBytes` is only used for disk sizes, not for memory. Memory is passed around as MiB, which should fit in an uint. Casting from uint to uint64 should not trigger overflow warnings (?). I tend to see `uint` as a more generic/flexible type, while `uint64` is a more specific/specialized one, which is why I favour the use of `uint` if we can.
If it leads to more complicated code, then sure, `uint64` can be used.

Reply:
> `ConvertGiBToBytes` is only used for disk sizes, not for memory. Memory is passed around as MiB, which should fit in an uint. Casting from uint to uint64 should not trigger overflow warnings (?). I tend to see `uint` as a more generic/flexible type, while `uint64` is a more specific/specialized one, which is why I favour the use of `uint` if we can. If it leads to more complicated code, then sure, `uint64` can be used.

Yes, `ConvertGiBToBytes` is for disk size (my mistake). memory is casted to `uint64` as part of telemetry data https://github.com/praveenkumar/crc/commit/ef1b33c3fd99c0f8d872c68d8dd968d17b068193#diff-850d36379066431f373c32a3db8d3bffb44ed8300d324f592d94b8bcab2dd500L268 .

Reply:
Also validation side https://github.com/praveenkumar/crc/commit/ef1b33c3fd99c0f8d872c68d8dd968d17b068193#diff-0541f68c893c48b4aa018a34aed7ee4bb96b95b067522e5925e3930f2d5acc47R54 `memory.TotalMemory()` return uint64 so it need again conversion. Now 2 different place need conversion.

Reply:
I don't think `uint` -> `uint64` casts trigger gosec warnings, so this should work:
```
func ValidateEnoughMemory(value uint) error {
  var valueBytes uint64
  totalMemory := memory.TotalMemory()
  logging.Debugf("Total memory of system is %d bytes", totalMemory)
  valueBytes = value * 1024 * 1024
  if totalMemory < valueBytes {
  ...
```